clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/manifest/KubernetesManifest.java

```diff
@@ -51,7 +51,9 @@ public KubernetesManifest clone() {
 
   @JsonIgnore
   public KubernetesKind getKind() {
-    return KubernetesKind.fromString(getRequiredField(this, "kind"));
+    //using ApiVersion here allows a translation from a kind of NetworkPolicy in the manifest to something
+    //like  NetworkPolicy.crd.projectcalico.org for custom resources
+    return KubernetesKind.fromString(getRequiredField(this, "kind"), getApiVersion());
```

Comment:
This will result in registering a lot of different kinds when multiple cluster versions are used. It also means kind filtering logic won't work unless the user knows which version to expect

Reply:
I'm not sure that's the case since it only gets the api group out of the version rather than making the full version part of the kind.  I could clarify that by getting the group from the version and passing that instead of the version.

Is there an example of a place where kind filtering logic happens?  I can check that to make sure that isn't affected by introducing this change from an unscoped kind in the manifest to a scoped kind in the cache.

Reply:
The api group is added to the `name` of the kind so equality works the same as it had before.  By passing the api group here we are able to construct the scoped name to use when [searching for the kind in the static list of known kinds](https://github.com/cerner/clouddriver/blob/4d8dacb1731f246825a140677943e59c14e266a3/clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/manifest/KubernetesKind.java#L155) so that we don't end up creating a new kind when a manifest comes in with just `PrometheusRule` as the kind.  The assumption here is that `PrometheusRule.monitoring.coreos.com` has already been dynamically registered by the caching agent.